File: 20-Apr-Java-7.30-Mihir/Classwork/LectureFlow/InheritanceDemo.java
import java.util.Scanner;

/*
 * Inheritance : The object of one class acquiring the properties of
 * 				 object of another class is called Inheritance.
 * 
 * 				Creating a new class from an existing class is called Inheritance.
 * 
 * Types of Inheritance
 * 
 * 1) Single Inheritance
 * 2) Multilevel Inheritance
 * 3) Multiple Inheritance
 * 4) Hierarchical Inheritance
 * 5) Hybrid Inheritance*  * 
 * 
 * * 
 */


class A{
	
	int a;
	Scanner sc = new Scanner(System.in);
	
	void getA() {
		System.out.println("Enter A :");
		a = sc.nextInt();
	}
	
	void putA() {
		System.out.println("A = "+a);
	}
	
}


class B extends A{
	
	int b;
	Scanner sc = new Scanner(System.in);
	
	void getB() {
		System.out.println("Enter B :");
		b = sc.nextInt();		
	}
	
	void putB() {
		System.out.println("B = "+b);
	}
	
}

class C extends B{
	int c;
	Scanner sc = new Scanner(System.in);
	
	void getC() {
		System.out.println("Enter C :");
		c = sc.nextInt();		
	}
	
	void putC() {
		System.out.println("C = "+c);
	}
}


public class InheritanceDemo {

	public static void main(String[] args) {
		
		C c = new C();
		c.getA();
		c.getB();
		c.getC();
		
		c.putA();
		c.putB();
		c.putC();
		
	}
}

File: 5-Apr-C-C++/C++ Lecture Flow/24-5-22/inheritancedemo.cpp
/*
Inheritance :- 

The child class object acquiring the properties of parent class is called inheritance.

simple & practical definition:  creating a new class from an existing class.

Parent Class -> Base Class & Super Class.
Child Class  -> Derived Class & Sub Class.

5 types of inheritance

1) Single-level inheritance
2) Multilevel inheritance
3) Multiple inheritance
4) Hierarchical inheritance
5) hybrid inheritance


// Object will always be created of derived class in inheritance.
*/


#include<iostream>
using namespace std;

// Base Class
class A : public B{          
	
	public:
		int a;
		
		void getA()
		{
			cout<<"Enter A : ";
			cin>>a;
		}
		
		void printA()
		{
			cout<<"A = "<<a;
		}
};

// Derived Class

class B : public A{
	
	public:
			int b;
		
			void getB()
			{
				cout<<"\nEnter B : ";
				cin>>b;
			}
			
			
			void printB()
			{
				cout<<", B = "<<b;
			}
};

int main()
{
	B y;
	y.getA();
	y.getB();
	y.printA();
	y.printB();
		
	
	return 0;
}

